plugin/storage/grpc/proto/storage_v1/storage_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage_v1_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/jaegertracing/jaeger/plugin/storage/grpc/proto/storageprototest"
+	"github.com/jaegertracing/jaeger/plugin/storage/grpc/proto/storage_v1"
+	"github.com/jaegertracing/jaeger/model"
+)
+
+func TestGetTraceRequestMarshalProto(t *testing.T) {
+	testCases := []struct {
+		name      string
+		marshal   func(proto.Message) ([]byte, error)
+		unmarshal func([]byte, proto.Message) error
+		expected  string
+	}{
+		{
+			name:      "protobuf",
+			marshal:   proto.Marshal,
+			unmarshal: proto.Unmarshal,
+		},
+		{
+			name: "JSON",
+			marshal: func(m proto.Message) ([]byte, error) {
+				out := new(bytes.Buffer)
+				err := new(jsonpb.Marshaler).Marshal(out, m)
+				if err != nil {
+					return nil, err
+				}
+				return out.Bytes(), nil
+			},
+			unmarshal: func(in []byte, m proto.Message) error {
+				return jsonpb.Unmarshal(bytes.NewReader(in), m)
+			},
+			expected: `{"traceId":"AAAAAAAAAAIAAAAAAAAAAw=="}`,
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			ref1 := storage_v1.GetTraceRequest{TraceID: model.NewTraceID(2, 3)}
+			ref2 := storageprototest.GetTraceRequest{
+				// TODO: would be cool to fuzz that test
+				TraceId: []byte{0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 3},
+			}
+			d1, err := testCase.marshal(&ref1)
+			require.NoError(t, err)
+			d2, err := testCase.marshal(&ref2)
+			require.NoError(t, err)
+			assert.Equal(t, d2, d1)
+			if testCase.expected != "" {
+				assert.Equal(t, testCase.expected, string(d1))
+			}
+			// test unmarshal
+			var ref1u storage_v1.GetTraceRequest
+			err = testCase.unmarshal(d2, &ref1u)
+			require.NoError(t, err)
+			assert.Equal(t, ref1, ref1u)
+		})
+	}
+}
+
+func TestFindTraceIDsResponseMarshalProto(t *testing.T) {
```

Comment:
other than serializing a different type, what does this test validate?  The embedded type is still `TraceID` as in the test above. It would be more useful to test serialization of `SpansResponceChunk`, because it uses an imported type, which is different from custom type `TraceID` which is just `bytes` in proto.

Reply:
I don't really see an effective way to test `SpansResponseChunk`, creating a type in the _test proto file would contain the same import (just missing the notnull annotation). All that I can think to do is marshal and then unmarshal a `SpansResponseChunk` and check that before == after.

Reply:
yes, to-from is what I meant, not the types of tests done with _test.proto (those are only useful for custom types with custom marshaling methods).

Reply:
OK I see, interestingly the `timestamp` field on `model.Span` seems to lose the `loc` field value when marshaled/unmarshaled. I'm trying to work out if that's a gogoproto issue.

Reply:
That may be expected. The timestamps we capture are absolute (UTC), the locale is not supposed to be preserved. We have `span.NormalizeTimestamps()` helper to deal with that.